fuzzbench/test_e2e_run.py

```diff
@@ -26,6 +28,15 @@
 class TestEndToEndRunResults:
     """Checks the result of a test experiment run."""
 
+    def test_jobs_dependency(self):
+        """Tests that jobs dependency preserves during working."""
+        jobs = {
+            name: Job.fetch(name, connection=redis.Redis(host='queue-server'))
+            for name in ['base-image', 'base-runner', 'base-builder']
+        }
+        assert jobs['base-image'].ended_at <= jobs['base-runner'].started_at
+        assert jobs['base-image'].ended_at <= jobs['base-builder'].started_at
```

Comment:
base-runner started_at <= base-builder's started at as well.

Reply:
According to current build dependency logic, it is not required.
Wdyt? @lszekeres @jonathanmetzman 
We can add this dependency logic into both #543 and here for consistency.

Reply:
Yeah I don't think this makes sense to test for. They are built in parallel.

Reply:
So, shall we change it back? Say, base-runner depends on base-image, not base-builder?

Reply:
yes, 
base-builder depends on base-image, and 
base-runner depends on base-image

Reply:
OK. Change it back now. Done.